pkg/ccl/crosscluster/logical/logical_replication_writer_processor.go

```diff
@@ -983,6 +987,9 @@ func (t *txnBatch) HandleBatch(
 	stats := batchStats{}
 	var err error
 	if len(batch) == 1 {
+		if t.spec.Discard == jobspb.LogicalReplicationDetails_DiscardAllDeletes && len(batch[0].KeyValue.Value.RawBytes) == 0 {
+			return stats, nil
+		}
```

Comment:
Wonder how you feel about moving some of the verbosity out into a func:

```golang
func (t *txnBatch) shouldDiscardDeletes() bool {
	return t.spec.Discard == jobspb.LogicalReplicationDetails_DiscardAllDeletes
}
```

I don't feel strongly feel free to ignore.